src/System.Private.CoreLib/shared/System/Char.cs

```diff
@@ -904,7 +904,14 @@ public static bool IsSurrogatePair(string s, int index)
 
         public static bool IsSurrogatePair(char highSurrogate, char lowSurrogate)
         {
-            return IsHighSurrogate(highSurrogate) && IsLowSurrogate(lowSurrogate);
+            // Since both the high and low surrogate ranges are exactly 0x400 elements
+            // wide, and since this is a power of two, we can perform a single comparison
+            // by baselining each value to the start of its respective range and taking
+            // the logical OR of them.
+
+            uint highSurrogateOffset = (uint)highSurrogate - CharUnicodeInfo.HIGH_SURROGATE_START;
+            uint lowSurrogateOffset = (uint)lowSurrogate - CharUnicodeInfo.LOW_SURROGATE_START;
+            return (highSurrogateOffset | lowSurrogateOffset) <= CharUnicodeInfo.HIGH_SURROGATE_RANGE;
```

Comment:
Nice improvements.  Do we have appropriate test coverage for these existing APIs in corefx?

Reply:
Yes, there's test coverage at https://github.com/dotnet/corefx/blob/5cdcb74c3ee82fcf99d6dee484b18681aab97126/src/System.Runtime/tests/System/CharTests.cs#L606.